src/box/vy_scheduler.c

```diff
@@ -734,6 +734,10 @@ vy_scheduler_wait_checkpoint(struct vy_scheduler *scheduler)
 			return -1;
 		}
 		fiber_cond_wait(&scheduler->dump_cond);
+		if (fiber_is_cancelled()) {
+			diag_set(FiberIsCancelled);
+			return -1;
```

Comment:
Better log an error in this case as we do for other errors.

Reply:
Done.

```diff
--- a/src/box/vy_scheduler.c
+++ b/src/box/vy_scheduler.c
@@ -730,17 +730,20 @@ vy_scheduler_wait_checkpoint(struct vy_scheduler *scheduler)
                        /* A dump error occurred, abort checkpoint. */
                        struct error *e = diag_last_error(&scheduler->diag);
                        diag_set_error(diag_get(), e);
-                       say_error("vinyl checkpoint failed: %s", e->errmsg);
-                       return -1;
+                       goto error;
                }
                fiber_cond_wait(&scheduler->dump_cond);
                if (fiber_is_cancelled()) {
                        diag_set(FiberIsCancelled);
-                       return -1;
+                       goto error;
                }
        }
        say_info("vinyl checkpoint completed");
        return 0;
+error:
+       say_error("vinyl checkpoint failed: %s",
+                 diag_last_error(diag_get())->errmsg);
+       return -1;
 }
```